lib/active_merchant/billing/gateways/d_local.rb

```diff
@@ -163,23 +163,28 @@ def add_card(post, card, action, options = {})
           post[:card][:network_token] = card.number
           post[:card][:cryptogram] = card.payment_cryptogram
           post[:card][:eci] = card.eci
-          # used case of Network Token: 'CARD_ON_FILE', 'SUBSCRIPTION', 'UNSCHEDULED_CARD_ON_FILE'
-          if options.dig(:stored_credential, :reason_type) == 'unscheduled'
-            if options.dig(:stored_credential, :initiator) == 'merchant'
-              post[:card][:stored_credential_type] = 'UNSCHEDULED_CARD_ON_FILE'
-            else
-              post[:card][:stored_credential_type] = 'CARD_ON_FILE'
-            end
-          else
-            post[:card][:stored_credential_type] = 'SUBSCRIPTION'
-          end
-          # required for MC debit recurrent in BR 'USED'(subsecuence Payments) . 'FIRST' an inital payment
-          post[:card][:stored_credential_usage] = (options[:stored_credential][:initial_transaction] ? 'FIRST' : 'USED') if options[:stored_credential]
         else
           post[:card][:number] = card.number
           post[:card][:cvv] = card.verification_value
         end
 
+        # used case of Network Token: 'CARD_ON_FILE', 'SUBSCRIPTION', 'UNSCHEDULED_CARD_ON_FILE'
```

Comment:
Could you add lines 171-180 to be inside `if options[:stored_credential]`? Because we only want to be sending those fields is `options[:stored_credential]` is present

Reply:
Looks like `stored_credential_type` is an option [here](https://github.com/activemerchant/active_merchant/pull/5112/files#diff-794e2c4393901725c7a286d8cd01de14e51e7ad9eb80370e34d91a879ddff305L71) If I make this change I will no longer get this [result](https://github.com/activemerchant/active_merchant/pull/5112/files#diff-794e2c4393901725c7a286d8cd01de14e51e7ad9eb80370e34d91a879ddff305L73) it is ok?

Reply:
That's looks like an incorrect test to me because in DLocal `stored_credential_type` is a field we look for in options if it makes sense but it's a DLocal field we send in a request. 'SUBSCRIPTION' looks be like a default value for any transaction that uses NT which could be an issue

Reply:
Ok, thanks!

Reply:
Good thing you were able to catch that we were always sending  SUBSCRIPTION for all NT transactions even if they aren't using stored credentials. Please do move lines 171-180 to be inside if `options[:stored_credential]` because we don't want to have any issues with us accidentally marking transactions as subscription even when they aren't

Reply:
Done!